Source dao test TestSourceSubcollectionWithUserOwnership() fix

petracihalova opened the pull request:
without JIRA ticket

sometimes when I run the tests I get this result
```
--- FAIL: TestSourceSubcollectionWithUserOwnership (2.16s)
    source_dao_test.go:1082: Expected source IDS [103 105] are not same with obtained ids: [105 103]
```

I found out that subcollection source IDs can be returned in random order so they have to be sorted before comparing them with the expected result

EDIT:
updated other tests that contains same logic

Commit:
Source DAO test fix

IDs can be returned in random order so they
have to be sorted before comparing them with the expected result

lindgrenj6 reviewed (approved):
...I can't wait until we have generic sort functions after go 1.18 so `sort.Ints` would work on these int64s.

Commit:
application DAO test fix

IDs can be returned in random order so they
have to be sorted before comparing them with the expected result

Commit:
authentication DAO test fix

IDs can be returned in random order so they
have to be sorted before comparing them with the expected result

Commit:
application authentication DAO test fix

IDs can be returned in random order so they
have to be sorted before comparing them with the expected result
